src/event-db/src/lib.rs

```diff
@@ -22,6 +20,19 @@ const DATABASE_URL_ENVVAR: &str = "EVENT_DB_URL";
 /// Must equal the last Migrations Version Number.
 pub const DATABASE_SCHEMA_VERSION: i32 = 9;
 
+/// Event database errors
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(" Schema in database does not match schema supported by the Crate. The current schema version: {was}, the schema version we expected: {expected}")]
+    MismatchedSchema { was: i32, expected: i32 },
+    #[error(transparent)]
+    TokioPostgresError(#[from] tokio_postgres::Error),
+    #[error(transparent)]
+    TokioPostgresRunError(#[from] RunError<tokio_postgres::Error>),
```

Comment:
This error type seems like it should be database agnostic (i.e. it appears in the trait, which is the same regardless of whether we use postgres, mongo, some in-memory thing, etc).

If that's the case, it might be worth having a single `Unknown(Report)` variant that is for "random database errors that we can't handle" (like connection errors), that way we're not leaking implementation details to the caller

Reply:
Fixed